src/com/amazon/ion/impl/lite/IonSymbolLite.java

```diff
@@ -308,8 +308,8 @@ private String stringValue(SymbolTableProvider symbolTableProvider)
             return null;
         }
         String name = _stringValue(symbolTableProvider);
-        if (name == null && _sid != 0) {
-            assert(_sid > 0);
+        if (name == null) {
```

Comment:
This was a bug. The documentation says that `stringValue` will thrown `UnknownSymbolException` for symbols with unknown text, and symbol 0 is a symbol with unknown text.